ansible/roles/satellite-manage-host/tasks/version_6.x.yml

```diff
@@ -0,0 +1,26 @@
+---
+
+- name: Get hosts for guid in organization
+  theforeman.foreman.foreman_search_facts:
+    username: "{{ satellite_admin }}"
+    password: "{{ satellite_admin_password }}"
+    server_url: "{{ satellite_url }}"
+    validate_certs: no
+    resource: hosts
+    organization: "{{ satellite_org }}"
+    search: '*{{ guid }}*'
```

Comment:
I have a concern here... we currently use 4 hexadecimal lowercase character GUIDs. With babylon we will be moving to 5 character guids with character set of any lowercase letter and number.

So at present we could get guid `feed` or `ab01`, but consider the hostnames `lab01-server.feed.example.com`, `lab-datafeed.ab01.example.com`.

It gets worse with babylon where the guid may be a string like `admin`.

It seems to me that this pattern search is not the right way to find satellite hosts. We should be able to determine more precisely the DNS domain that would be used by a lab such that we could do something more like:

`    search: '*.{{ guid }}.{{ satelite_host_domain }}'`

Reply:
@jkupferer made change in search value to `.{{guid}}.`